### PR TITLE
Add check for declining pings after the deadline

### PR DESCRIPTION
**Pull request by claudiu725**
The ping-pong smart contract was designed with a user limit in mind. Because there's no user limit, but only a max funding limit, the user limit is enforced only if a user can't deposit after the deadline, after other users have withdrawn their sum (and hence the smart contract balance has decreased).

### CODE REVIEW

File: contracts/examples/ping-pong-egld/src/lib.rs [hunk 1]
```diff
@@ -38,6 +38,11 @@ pub trait PingPong {
 			"smart contract not active yet"
 		);
 
+		require!(
+			self.get_block_nonce() < self.get_deadline(),
+			"smart contract already ended"
```
Comment (dorin-iancu) [inline contracts/examples/ping-pong-egld/src/lib.rs:43]: "smart contract already ended" doesn't sound too good. Maybe rephrase to "deadline has passed" or something similar.

**Review by dorin-iancu — DISMISSED** [review]

File: contracts/examples/ping-pong-egld/mandos/ping-pong-call-ping-after-deadline.scen.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                "to": "address:the_ping_pong_contract",
+                "value": "500,000,000,000",
+                "function": "ping",
+                "arguments": [],
+                "gasLimit": "100,000,000",
+                "gasPrice": "0"
+            },
+            "expect": {
+                "status": "4",
+                "message": "str:smart contract already ended",
```
Comment (dorin-iancu) [inline contracts/examples/ping-pong-egld/mandos/ping-pong-call-ping-after-deadline.scen.json:27]: Same here

**Review by dorin-iancu — DISMISSED** [review]

**Comment by claudiu725** [comment]
Signed the commits and re-pushed them

**Review by dorin-iancu — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]
